frontend/lib/src/components/elements/ArrowVegaLiteChart/ArrowVegaLiteChart.test.tsx

```diff
@@ -60,30 +63,36 @@ const getProps = (props: Partial<PropsWithHeight> = {}): PropsWithHeight => ({
   ...props,
 })
 
+const getCanvasContext = (): CanvasRenderingContext2D => {
+  const canvas = screen
+    .getByTestId("stArrowVegaLiteChart")
+    // eslint-disable-next-line testing-library/no-node-access
+    .querySelector("canvas") as HTMLCanvasElement
+
+  return canvas.getContext("2d") as CanvasRenderingContext2D
+}
+
 describe("VegaLiteChart Element", () => {
-  it("renders without crashing", () => {
+  it("renders without crashing", async () => {
     const props = getProps()
-    const wrapper = mount(<ArrowVegaLiteChart {...props} />)
 
-    expect(wrapper.find("StyledVegaLiteChartContainer").length).toBe(1)
+    await render(<ArrowVegaLiteChart {...props} />)
+    expect(screen.getByTestId("stArrowVegaLiteChart")).toBeInTheDocument()
   })
 
-  it("pulls default config values from theme", () => {
+  it("pulls default config values from theme", async () => {
     const props = getProps({ theme: mockTheme.emotion })
 
-    const wrapper = mount(<ArrowVegaLiteChart {...props} />)
-    // @ts-expect-error
-    const generatedSpec = wrapper.instance().generateSpec()
+    render(<ArrowVegaLiteChart {...props} />)
 
-    expect(generatedSpec.config.background).toBe(
-      mockTheme.emotion.colors.bgColor
-    )
-    expect(generatedSpec.config.axis.labelColor).toBe(
-      mockTheme.emotion.colors.bodyText
-    )
+    await new Promise(process.nextTick)
```

Comment:
Is there a reason why we need to await for the next tick?

Reply:
I can add a comment. Essentially this waits for any async code to resolve (aka the await statements that create the view).